R-package/tests/testthat/test_model_compatibility.R

```diff
@@ -73,12 +75,15 @@ test_that("Models from previous versions of XGBoost can be loaded", {
     model_xgb_ver <- m[2]
     name <- m[3]
 
-    if (endsWith(model_file, '.rds')) {
-      booster <- readRDS(model_file)
-    } else {
-      booster <- xgb.load(model_file)
-    }
-    predict(booster, newdata = pred_data)
-    run_booster_check(booster, name)
+    ## don't print C level compatibility warnings
```

Comment:
Please keep the compatibility warnings. The warnings are expected.

Reply:
This is exactly why I suppressed them, because they are expected. They just pollute the test output.

Reply:
For this test case, we can add `gives_warning()` to ensure that this test throws a warning. I can do that in a future pull request.

Reply:
I have inhibited only C level warnings here, the R level warnings are still there and testhat still complains. So it's redundat. But ok, I have removed for now. It's for another PR indeed.